src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEParameterSymbol.cs

```diff
@@ -656,6 +658,44 @@ internal override bool IsCallerMemberName
             }
         }
 
+        private int? _cachedCallerArgumentExpressionParameterIndex;
+
+        internal override int CallerArgumentExpressionParameterIndex
+        {
+            get
+            {
+                if (_cachedCallerArgumentExpressionParameterIndex.HasValue)
+                {
+                    return _cachedCallerArgumentExpressionParameterIndex.Value;
+                }
+
+                var info = _moduleSymbol.Module.FindTargetAttribute(_handle, AttributeDescription.CallerArgumentExpressionAttribute);
+                var discardedUseSiteInfo = CompoundUseSiteInfo<AssemblySymbol>.Discarded;
+                bool isCallerArgumentExpression = !HasCallerLineNumberAttribute
+                    && !HasCallerFilePathAttribute
+                    && !HasCallerMemberNameAttribute
+                    && info.HasValue
+                    && new TypeConversions(ContainingAssembly).HasCallerInfoStringConversion(this.Type, ref discardedUseSiteInfo);
+
+                if (isCallerArgumentExpression)
+                {
+                    _moduleSymbol.Module.TryExtractStringValueFromAttribute(info.Handle, out var parameterName);
+                    var parameters = ContainingSymbol.GetParameters();
+                    for (int i = 0; i < parameters.Length; i++)
+                    {
+                        if (parameters[i].Name == parameterName)
+                        {
+                            _cachedCallerArgumentExpressionParameterIndex = i;
```

Comment:
>_cachedCallerArgumentExpressionParameterIndex = i; [](start = 28, length = 50)

This assignment is not guaranteed to be atomic, therefore, the implementation of this property is not thread safe. That is why we don't use ```Nullable<T>``` for "lazy" members. For example, we use ```ThreeState``` instead of ```bool?```. It looks like we can use a simple ```int``` field with initial value (-2), for example, to indicate uninitialized state.